tests/robot-tests/tests/libs/admin-common.robot

```diff
@@ -9,6 +9,25 @@ ${ANALYST1_BROWSER}=    analyst1
 
 
 *** Keywords ***
+user logs in via identity provider
```

Comment:
I think `get_auth_tokens.py` shares a lot of logic with this keyword. Potentially it could be modified to be used  in both robot tests and generic python test code

Reply:
Yeah I was surprised initially not to find any actual "real" login code in Robot itself, but I knew there was a bunch in Python code and also in the Puppeteer code too.  Would be nice to consolidate the Python code in some way - as you say, perhaps as a part of another piece of work.